client/lib/menu-data/menu-data.js

```diff
@@ -411,7 +434,47 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
 			this.change( { reset: true } );
 			this.emit( 'saved' );
 			callback && callback( null, parsedMenu );
-		} );
+		}
+	);
+}
+
+MenuData.prototype.saveMenu = function( menu, callback ) {
+	if ( ! menu ) {
+		menu = this.find( { id: this.lastChangedMenuID } );
+	}
+
+	if ( ! this.isValidMenu( menu ) ) {
+		callback && callback( new Error( 'Invalid menu' ) );
+		debug( 'saveMenu: fail' );
+		return;
+	}
+
+	menu = cloneDeep( menu );
+
+	if ( menu.id === this.getDefaultMenuId() ) {
+		return this.saveDefaultMenu();
+	}
+
+	this.restoreServerIDs( menu );
+
+	debug( 'saveMenu', menu );
+
+	const self = this;
+
+	/*
+	Because of how postEditStore works, we need to fire promises one after another.
+	 */
```

Comment:
The iterated page creation process below is not super easy to read. Maybe this comment could be slightly clearer and explain that _pages_ need to be created serially ("fire promises" is a little vague, IMO).

Aside: is this the right formatting style? For this kind of comment I'd have gone with two lines of `//`-prefixed comments.